tests/semantic_analyser.cpp

```diff
@@ -254,6 +262,33 @@ kprobe:f { fake }
   test(feature, "k:f { jiffies }", 1);
 }
 
+TEST(semantic_analyser, builtin_variables_inline)
+{
+  auto bpftrace = get_mock_bpftrace();
+  ConfigSetter configs{ bpftrace->config_, ConfigSource::script };
+  configs.set(ConfigKeyBool::probe_inline, true);
+
+  // Check argument builtins are rejected when `probe_inline` is enabled.
+  test_error(*bpftrace, "uprobe:/bin/sh:f { arg0 }", R"(
+stdin:1:20-24: ERROR: The arg0 builtin can only be used when the probe_inline config is disabled.
+uprobe:/bin/sh:f { arg0 }
+                   ~~~~
+)");
+  test_error(*bpftrace, "uprobe:/bin/sh:f { sarg0 }", R"(
+stdin:1:20-25: ERROR: The sarg0 builtin can only be used when the probe_inline config is disabled.
```

Comment:
So will this test go away when we fully kill `sarg`?

Reply:
Yes.